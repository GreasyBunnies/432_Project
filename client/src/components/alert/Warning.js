import React from 'react';
import {
    Alert
} from 'react-bootstrap';

class Warning extends React.Component {
    render() {
        return (
            <Alert variant="warning">
                <Alert.Heading>{this.props.header}</Alert.Heading>
                {this.props.message}
            </Alert>
        )
    }
}

export default Warning;
