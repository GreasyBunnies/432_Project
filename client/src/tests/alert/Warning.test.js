import React from 'react';
import ReactDOM from 'react-dom';

import Warning from '../../components/alert/Warning.js';

import 'jest-localstorage-mock';
import renderer from 'react-test-renderer';

describe("Rendering Warning alert", () => {
	it('Renders Warning without crashing', () => {
		const div = document.createElement('div');

		ReactDOM.render(<Warning />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
});

describe("Rendering Unavailable alert with message", () => {
	it('Displays the correct message', () => {
		const component = renderer.create(
			<Warning header="Test" message="This is a test"></Warning>,
		);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
