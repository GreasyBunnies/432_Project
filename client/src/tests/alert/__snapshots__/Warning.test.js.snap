// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Unavailable alert with message Displays the correct message 1`] = `
<div
  className="fade alert alert-warning show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    Test
  </div>
  This is a test
</div>
`;
